indico/modules/events/registration/client/js/components/RegistrationTagsEditableList.jsx

```diff
@@ -0,0 +1,151 @@
+
+  async function editRegistrationTags({add = [], remove = []}) {
+    try {
+      await indicoAxios.post(registrationTagAssign({reg_form_id: regformId, event_id: eventId}), {
+        registration_id: [registrationId],
+        add,
+        remove,
+      });
+    } catch (error) {
+      return handleAxiosError(error);
+    }
+  }
+
+  function onRemove(tag) {
+    setTags(tags.filter(t => t.id !== tag.id));
+    editRegistrationTags({remove: [tag.id]});
+  }
+
+  function onAdd(tag) {
+    const newTags = _.sortBy([...tags, tag], [t => t.title.toLowerCase()], 'title');
+    setTags(newTags);
+    if (newTags.length === allTags.length) {
+      setOpen(false);
+    }
+    editRegistrationTags({add: [tag.id]});
+  }
+
+  const trigger = (
+    <IButton disabled={unassignedTags.length === 0}>
+      <Translate>Add</Translate>
+      <Icon name="caret down" style={{marginRight: 0}} />
+    </IButton>
+  );
+
+  return (
+    <>
+      <div className="icon icon-tag" />
+      <div className="text">
+        <div className="label">
+          <Translate>Tags</Translate>
+        </div>
+        <div
+          styleName="removable-tags-container"
+          style={{marginTop: tags.length === 0 ? '0' : '3px'}}
+        >
+          {tags.length === 0 ? (
+            <Translate>No tags assigned.</Translate>
+          ) : (
+            tags.map(tag => (
+              <span
+                key={tag.title}
+                styleName="registration-tag removable"
+                style={{backgroundColor: tag.color}}
+              >
+                <span>{tag.title}</span>
+                <Icon name="close" onClick={() => onRemove(tag)} />
+              </span>
+            ))
+          )}
+        </div>
+      </div>
+      <div className="toolbar">
+        {unassignedTags.length === 0 ? (
+          trigger
+        ) : (
+          <PopoverDropdownMenu
+            onOpen={() => setOpen(true)}
+            onClose={() => setOpen(false)}
+            trigger={trigger}
+            open={open}
+            placement="right-start"
+          >
+            <Header size="small" textAlign="center">
+              <Translate>Add tags</Translate>
+            </Header>
+            <Divider style={{margin: 0}} />
+            <div styleName="selectable-tags-container">
+              {unassignedTags.map(tag => (
+                <div key={tag.title} onClick={() => onAdd(tag)}>
+                  <div
+                    styleName="registration-tag selectable"
+                    style={{
+                      backgroundColor: tag.color,
+                    }}
+                  >
+                    {tag.title}
+                  </div>
+                </div>
+              ))}
+            </div>
+            <Button
+              size="mini"
+              fluid
+              styleName="close-button"
+              content={Translate.string('Close')}
+              onClick={() => setOpen(false)}
+            />
```

Comment:
Do we need that close button? Clicking outside the popover does the job, and I think it's not particularly common for this kind of UI to have a dedicated close button (similar dropdowns in other places in Indico don't have one either).

Reply:
I suppose not, I mostly reused `PopoverDropdownMenu` from room booking which has a very similar behavior:
![image](https://user-images.githubusercontent.com/8739637/135074498-2e692f74-fdee-4c4f-acd4-5519df79b398.png)

Reply:
ah, I wasn't aware this one also had a cancel button. hm... @pferreir - any suggestions since you added the one in RB?

Reply:
Should I remove it?

What it looks like with and without:

![image](https://user-images.githubusercontent.com/8739637/135082070-40f2a4b7-4e38-463b-919e-f79c228ef46f.png)
![image](https://user-images.githubusercontent.com/8739637/135081993-42d22229-dc14-49ce-a927-1997fe7d8fc6.png)

Reply:
IMHO it looks better without